test/run_pass/test_module_cache.js

```diff
@@ -22,8 +22,8 @@ module_cache = require('run_pass/require1/module_cache.js');
 assert.equal(module_cache.i, 100);
 
 {
-  var test1 = require('run_pass/require1/node_modules/test_pkg');
-  var test2 = require('run_pass/require1/node_modules/test_index');
+  var test1 = require('run_pass/require1/test_pkg');
```

Comment:
This part is already covered by the introduced `test_module_require.js`, so I've removed it. The invalid modules part is moved into `test_module_require.js` (because I think it doesn't belong to the module caching).